File: 0686-repeated-string-match/solution.java
class Solution {
    public int repeatedStringMatch(String a, String b) {
        int ans=1;
        String str= new String(a);
        while(str.length()<b.length()){
            str+=a;
            ans++;
        }
        if(str.contains(b)){
            return ans;
        }
        str+=a;
        ans++;
        if(str.contains(b)){
            return ans;
        }
        return -1;
    }
}

File: 0686-repeated-string-match/solution.cpp
class Solution {
public:
    int recur(string a,string b,string str,int c){
        if(str.contains(b)){
            return c;
        }
        if(str.length() > b.length() ){
            str=str+a;
            c++;
            if(str.contains(b)){
                return c;
            }else{
                return -1;
            }
        }
        return recur(a,b,str+a,c+1);

    }
    int repeatedStringMatch(string a, string b) {
        return recur(a,b,a,1);
    }
};
